openmc/model/funcs.py

```diff
@@ -0,0 +1,578 @@
+    # Expanding out all terms and grouping according to what Quadric expects
+    # gives the following coefficients.
+    kwargs['a'] = dy*dy + dz*dz
+    kwargs['b'] = dx*dx + dz*dz
+    kwargs['c'] = dx*dx + dy*dy
+    kwargs['d'] = -2*dx*dy
+    kwargs['e'] = -2*dy*dz
+    kwargs['f'] = -2*dx*dz
+    kwargs['g'] = 2*(cy*dz - cz*dy)
+    kwargs['h'] = 2*(cz*dx - cx*dz)
+    kwargs['j'] = 2*(cx*dy - cy*dx)
+    kwargs['k'] = cx*cx + cy*cy + cz*cz - (dx*dx + dy*dy + dz*dz)*r*r
+
+    return Quadric(**kwargs)
+
+
+def subdivide(surfaces):
+    """Create regions separated by a series of surfaces.
+
+    This function allows regions to be constructed from a set of a surfaces that
+    are "in order". For example, if you had four instances of
+    :class:`openmc.ZPlane` at z=-10, z=-5, z=5, and z=10, this function would
+    return a list of regions corresponding to z < -10, -10 < z < -5, -5 < z < 5,
+    5 < z < 10, and 10 < z. That is, for n surfaces, n+1 regions are returned.
+
+    Parameters
+    ----------
+    surfaces : sequence of openmc.Surface
+        Surfaces separating regions
+
+    Returns
+    -------
+    list of openmc.Region
+        Regions formed by the given surfaces
+
+    """
+    regions = [-surfaces[0]]
+    for s0, s1 in zip(surfaces[:-1], surfaces[1:]):
+        regions.append(+s0 & -s1)
+    regions.append(+surfaces[-1])
+    return regions
+
+
+def pin(surfaces, items, subdivisions=None, divide_vols=True,
+        **kwargs):
+    """Convenience function for building a fuel pin
+
+    Parameters
+    ----------
+    surfaces : iterable of :class:`openmc.Cylinder`
+        Cylinders used to define boundaries
+        between items. All cylinders must be
+        concentric and of the same orientation, e.g.
+        all :class:`openmc.ZCylinder`
+    items : iterable
+        Objects to go between ``surfaces``. These can be anything
+        that can fill a :class:`openmc.Cell`, including
+        :class:`openmc.Material`, or other :class:`openmc.Universe`
+        objects. There must be one more item than surfaces,
+        which will span all space outside the final ring.
+    subdivisions : None or dict of int to int
+        Dictionary describing which rings to subdivide and how
+        many times. Keys are indexes of the annular rings
+        to be divided. Will construct equal area rings
+    divide_vols : bool
+        If this evaluates to ``True``, then volumes of subdivided
+        :class:`openmc.Material` instances will also be divided by the
+        number of divisions.  Otherwise the volume of the
+        original material will not be modified before subdivision
+    kwargs:
+        Additional key-word arguments to be passed to
+        :class:`openmc.Universe`, like ``name="Fuel pin"``
+
+    Returns
+    -------
+    :class:`openmc.Universe`
+        Universe of concentric cylinders filled with the desired
+        items
+    """
+    if "cells" in kwargs:
+        raise SyntaxError(
```

Comment:
I would change this to be a `ValueError` as it's not really a syntax error and `SyntaxError`s are typically reserved for the Python interpreter.